src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -352,8 +352,9 @@ export default class ChangeElementTemplateHandler {
         }
       }
 
-      // (3) add new inputs and outputs (unless optional)
-      else if (shouldUpdate(newPropertyValue, newProperty)) {
+      // (3) add new inputs and outputs (unless optional or undefined value)
+      else if (newPropertyValue && shouldUpdate(newPropertyValue, newProperty)) {
```

Comment:
Should we test for existence of `newPropertyValue` here? It could be an empty string, too.